### ISSUE TITLE
Fix `at-rule-descriptor-no-unknown` false positives for page and margin context properties in `@page`

### ISSUE THREAD

**Issue by steffenweber** [stylelint/stylelint#8316]
### What minimal example or steps are needed to reproduce the bug?

```css
@page {
	margin: 2cm;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAhgcxgAmAHQDs8AXAWzQCcMBLA+bAJjFIG5CBfEAGhADNqANjAByaUnEQwAHmJRCAdGADOS7uAgF+GBCHwFs2PCGnEYBKEqP0A2oQMGjS4gE8hA2sQC0kTdQyenNHNKKCM7bABdDjUfLQAxCApyYh0AKyUNNVgUVUQ9BxAnVxh3AmIrQxABNFMnIy5wxxc3D28NLQDiIKgQiqNq2vKQaPYgA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.13.2",
  "stylelint-config-standard": "37.0.0"
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problems were reported: 

```
2:2-8 error Unexpected unknown descriptor "margin" for at-rule "@page" at-rule-descriptor-no-unknown
```

### Do you have a proposal to fix the bug?

This seems to be valid CSS according to https://developer.mozilla.org/en-US/docs/Web/CSS/@page

**Comment by jeddy3** [comment]
@steffenweber Thanks for the report and for using the template.

As well its [four descriptors](https://csstree.github.io/docs/syntax/#Atrule:page), it seems `@page` accepts [specific _page properties_](https://drafts.csswg.org/css-page/#page-property-list), too.

We'll need to account for those. It'll involve looking into whether we should address this upstream in CSSTree or here in Stylelint.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by Mouvedia** [comment]
`css-tree`—which we are depending on—is only listing 4 descriptors on [this page](https://csstree.github.io/docs/syntax/#Atrule:page) which `margin` is not part of.
This is probably generated from https://www.w3.org/TR/css-page-3/#page-descriptor-table which list the same 4.
i.e. `bleed`, `marks`, `page-orientation` and `size`

In short, we should probably ignore [these properties](https://www.w3.org/TR/css-page-3/#page-property-list).
`margin` happens to be part of that list.

---

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ryo-manba** [comment]
It seems tricky to validate descriptors in at-rules that can also accept regular properties.
Unlike [@container](https://csstree.github.io/docs/syntax/#Atrule:container)  or [@layer](https://csstree.github.io/docs/syntax/#Atrule:layer), which have no descriptor declarations, [@page](https://csstree.github.io/docs/syntax/#Atrule:page) has both descriptors and [page properties](https://drafts.csswg.org/css-page/#page-properties).

Maybe we should skip descriptor validation for `@page`. What do you think?

**Comment by jeddy3** [comment]
As _page properties_ is a not very long definitive list, I think we can add a special case of those inside of `@page`.

**Comment by Mouvedia** [comment]
Either limit it to at-rules that only include descriptors which could be very unexpected for users even if we add a note on the README. Or we can just simply ignore the properties that are listed as supported by `@page`. I lean towards the latter.

We will have to be thorough and check each at-rules though.
i.e. `@page` is probably not the only one
